lib/label.ml

```diff
@@ -67,14 +67,20 @@ module Relation = struct
       with Not_found -> (s, None)
 end
 
+type non_det_mode =
```

Comment:
I think this should be `non_det`. The mode is a concept from `Block` and is represented by an other type.